src/include/processor/operator/base_shortest_path.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "queue"
+
+#include "processor/operator/physical_operator.h"
+#include "storage/storage_structure/lists/lists.h"
+
+namespace kuzu {
+namespace processor {
+
+struct BFSState {
```

Comment:
I don't think you should keep children, listSyncState, listHandle, etc. in the state. For each node, its state can simply be as follows:
```
struct NodeState {
    uint64_t relID = UINT64_MAX;
    node_offset_t predecessor = UINT64_MAX; // should change to nodeID_t eventually.
}

```